src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisWorkerService.java

```diff
@@ -178,56 +181,122 @@ private static SensitivityAnalysisParameters buildParameters(SensitivityAnalysis
-                context.getSensitivityAnalysisInputs().getVariablesSets(),
-                sensitivityAnalysisParameters,
-                sensitivityAnalysisExecutionService.getLocalComputationManager(),
-                reporter);
-            if (resultUuid != null) {
-                futures.put(resultUuid, future);
-            }
-            return future;
+            return runner.runAsync(context, sensitivityAnalysisRunner, resultUuid, reporter, network, sensitivityAnalysisParameters);
         } finally {
             lockRunAndCancelSensitivityAnalysis.unlock();
         }
     }
 
+    private CompletableFuture<SensitivityAnalysisResult> runAsyncInMemory(SensitivityAnalysisRunContext context,
+                                                                          SensitivityAnalysis.Runner sensitivityAnalysisRunner,
+                                                                          UUID resultUuid,
+                                                                          Reporter reporter,
+                                                                          Network network,
+                                                                          SensitivityAnalysisParameters parameters) {
+        List<SensitivityFactor> factors = context.getSensitivityAnalysisInputs().getFactors().stream().flatMap(Collection::stream).toList();
+        List<Contingency> contingencies = new ArrayList<>(context.getSensitivityAnalysisInputs().getContingencies());
+
+        SensitivityFactorReader sensitivityFactorReader = new SensitivityFactorModelReader(factors, network);
+        SensitivityResultModelWriter writer = new SensitivityResultModelWriter(contingencies);
+
+        CompletableFuture<Void> future = sensitivityAnalysisRunner.runAsync(
+            network,
+            context.getVariantId() != null ? context.getVariantId() : VariantManagerConstants.INITIAL_VARIANT_ID,
+            sensitivityFactorReader,
+            writer,
+            contingencies,
+            context.getSensitivityAnalysisInputs().getVariablesSets(),
+            parameters,
+            sensitivityAnalysisExecutionService.getLocalComputationManager(),
+            reporter);
+        return future.thenApply(r -> new SensitivityAnalysisResult(factors, writer.getContingencyStatuses(), writer.getValues()));
+    }
+
+    private CompletableFuture<Void> runAsyncPersisted(SensitivityAnalysisRunContext context,
+                                                      SensitivityAnalysis.Runner sensitivityAnalysisRunner,
+                                                      UUID resultUuid,
+                                                      Reporter reporter,
+                                                      Network network,
+                                                      SensitivityAnalysisParameters parameters) {
+        List<List<SensitivityFactor>> groupedFactors = context.getSensitivityAnalysisInputs().getFactors();
```

Comment:
That's actually the problem of SensitivityFactorModelReader, by construction it needs the whole data to be created so in itself it doesn't solve any memory issue on the factors... We talked about it with Geoffroy and with the tests I did, I think it should be fine in terms of memory consumption. But if we want to avoid this issue we need to implement another SensitivityFactorReader. And even if we implement that, if I remember correctly @geofjamg  told me that in current powsyl computation model the SensitivityFactorReader is read all at once, so even if we fix it in the implementation, still the whole object will be loaded by powsybl... So this problem is a trickier one.